### PR TITLE
RestoreDashboards: Create trash section

### LINKED ISSUE #87393: RestoreDashboards: Create trash section
This issue is for creating a new section in dashboards where deleted dashboards will be stored for a while to make restoring them easier. 
320806074-3d6f61fe-d2c8-43a8-a095-c87c7b296d05

```[tasklist]
### Tasks
- [x] Create trash section
- [ ] Add browse view
```

### PR DESCRIPTION
**Pull request by L-M-K-B**
**What is this feature?**
This PR creates a new page for a trash section in dashboards. The page is only showing if the feature toggle `dashboardRestore` in set to true and the user has the right to delete dashboards.

**Which issue(s) does this PR fix?**:
Fixes #87393

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/core/utils/navBarItem-translations.ts [hunk 1]
```diff
@@ -40,6 +40,8 @@ export function getNavTitle(navId: string | undefined) {
       return t('nav.reporting.title', 'Reporting');
     case 'dashboards/public':
       return t('nav.public.title', 'Public dashboards');
+    case 'dashboards/trash':
```
Comment (eledobleefe) [inline public/app/core/utils/navBarItem-translations.ts:43]: I think you forgot to mark the subtitle for translation.

**Review by eledobleefe — APPROVED** [review]
LGTM 🙌

File: pkg/services/featuremgmt/toggles_gen.json [hunk 2]
```diff
@@ -2250,6 +2245,30 @@
...
+      "metadata": {
+        "name": "disableSSEDataplane",
+        "resourceVersion": "1716564259132",
+        "creationTimestamp": "2024-05-24T15:24:19Z"
+      },
+      "spec": {
+        "description": "Disables dataplane specific processing in server side expressions.",
+        "stage": "experimental",
+        "codeowner": "@grafana/observability-metrics"
+      }
```
Comment (tskarhed) [inline pkg/services/featuremgmt/toggles_gen.json:2271]: You might want to do something about this part

File: go.work.sum [hunk 3]
```diff
@@ -749,6 +749,7 @@ github.com/kataras/tunnel v0.0.4/go.mod h1:9FkU4LaeifdMWqZu7o20ojmW4B7hdhv2CMLwf
 github.com/kelseyhightower/envconfig v1.4.0 h1:Im6hONhd3pLkfDFsbRgu68RDNkGF1r3dvMUtDTo2cv8=
 github.com/kisielk/errcheck v1.5.0 h1:e8esj/e4R+SAOwFwN+n3zr0nYeCyeweozKfO23MvHzY=
 github.com/kisielk/gotool v1.0.0 h1:AV2c/EiW3KqPNT9ZKl07ehoAGi4C5/01Cfbblndcapg=
+github.com/kisielk/sqlstruct v0.0.0-20201105191214-5f3e10d3ab46 h1:veS9QfglfvqAw2e+eeNT/SbGySq8ajECXJ9e4fPoLhY=
```
Comment (tskarhed) [inline go.work.sum:752]: This change isn't related to this PR

**Review by tskarhed — COMMENTED** [review]
All the related changes look good, but you might need to do some git magic to remove the unrelated changes

File: pkg/services/featuremgmt/toggles_gen.json (hunk 2 above)
Comment (L-M-K-B) [reply pkg/services/featuremgmt/toggles_gen.json:2271]: 🤔 Don't know how this got in there but I assume it's coming from me merging main to solve a conflict in this file... I'll check this.

Comment (L-M-K-B) [reply pkg/services/featuremgmt/toggles_gen.json:2271]: @tskarhed These two are actually not newly added but they were moved from above (beginning line 2206). So I'm assuming it's alright like this.

File: pkg/services/featuremgmt/toggles_gen.json [hunk 4]
```diff
@@ -2264,6 +2259,30 @@
...
+      "metadata": {
+        "name": "influxdbRunQueriesInParallel",
+        "resourceVersion": "1716816471156",
+        "creationTimestamp": "2024-05-27T13:27:51Z"
+      },
+      "spec": {
+        "description": "Enables running InfluxDB Influxql queries in parallel",
+        "stage": "privatePreview",
+        "codeowner": "@grafana/observability-metrics"
+      }
```
Comment (tskarhed) [inline pkg/services/featuremgmt/toggles_gen.json:2285]: Are these still supposed to be here? Is something out of sync with main?

**Review by tskarhed — APPROVED** [review]

**Pull request merged**